internal/infrastructure/kubernetes/infra_client.go

```diff
@@ -23,26 +25,27 @@ func New(cli client.Client) *InfraClient {
 }
 
 func (cli *InfraClient) CreateOrUpdate(ctx context.Context, key client.ObjectKey, current client.Object, specific client.Object, updateChecker func() bool) error {
-	if err := cli.Client.Get(ctx, key, current); err != nil {
-		if kerrors.IsNotFound(err) {
-			// Create if it does not exist.
-			if err := cli.Client.Create(ctx, specific); err != nil {
-				return err
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		if err := cli.Client.Get(ctx, key, current); err != nil {
+			if kerrors.IsNotFound(err) {
+				// Create if it does not exist.
+				if err := cli.Client.Create(ctx, specific); err != nil {
+					return errors.Wrap(err, "for Create")
+				}
 			}
-		}
-	} else {
-		// Since the client.Object does not have a specific Spec field to compare
-		// just perform an update for now.
-		if updateChecker() {
-			specific.SetResourceVersion(current.GetResourceVersion())
-			specific.SetUID(current.GetUID())
-			if err := cli.Client.Update(ctx, specific); err != nil {
-				return err
+		} else {
+			// Since the client.Object does not have a specific Spec field to compare
+			// just perform an update for now.
+			if updateChecker() {
+				specific.SetUID(current.GetUID())
```

Comment:
can you add `specific.SetResourceVersion(current.GetResourceVersion())` back?

Reply:
I am testing the `resourceVersion` is unnecessary. And it will be updated on every editing.